src/rust/broker/src/server/ws.rs

```diff
@@ -34,61 +35,56 @@ async fn handle_websocket(socket: WebSocket, state: Arc<AppState>) {
     let r = async move {
         while let Some(Ok(msg)) = ws_rx.next().await {
             if let Ok(msg) = msg.to_text() {
-                log::info!("Got message ({}).", msg);
                 let new_sub: Result<Subscription, _> = serde_json::from_str(msg);
                 if let Ok(new_sub) = new_sub {
-                    log::info!("Subscription updated ({new_sub:?}).");
+                    debug!("Subscription updated ({new_sub:?}).");
                     *sub2.write().await = Some(new_sub);
                 } else {
-                    log::warn!("Got invalid JSON format from client, closing.");
+                    warn!("Got invalid JSON format from client, closing connection.");
                     break;
                 }
             } else {
-                log::warn!("Message sent by client is not text, closing.");
+                warn!("Message sent by client is not text, closing connection.");
                 break;
             }
         }
         let _ = ws_tx2.write().await.close().await;
-        log::warn!("Connection ended, closing.");
+        warn!("Connection ended.");
     };
 
     let t = async move {
         let sub = sub.clone();
         loop {
             let mut r = rx.recv().await;
             while matches!(r, Err(RecvError::Lagged(_))) {
-                log::warn!("Messages dropped due to lag.");
+                warn!("Messages dropped due to lag.");
                 r = rx.recv().await;
             }
             if let Ok(item) = r {
                 let s = sub.read().await;
-                for s in s.iter() {
+                if let Some(s) = &*s {
                     if is_match(s, &item) {
                         let item_str = serde_json::to_string(&item).unwrap();
                         if let Err(e) = ws_tx.write().await.send(Message::Text(item_str)).await {
-                            log::warn!("Could not send event to user: {}.", e);
+                            warn!("Could not send event to user: {}, closing connection.", e);
                             break;
                         } else {
-                            log::info!("Sent message.")
+                            debug!("Sent message.")
                         }
                     }
                 }
             } else {
-                log::error!("Got error: {}.", r.unwrap_err());
+                error!("Got unexpected error: {}.", r.unwrap_err());
                 break;
             }
         }
         let _ = ws_tx.write().await.close().await;
     };
 
     tokio::select! {
-        _ = t => {
-            log::warn!("Connection broken (t).")
-        }
-        _ = r => {
-            log::warn!("Connection broken (r).")
-        }
+        _ = t => {}
+        _ = r => {}
```

Comment:
Did we remove this because the error logs are handled elsewhere in every case now?

Reply:
Yes. Each case has its own internal logging now.